### PR TITLE
chore(deps): bump bridge-encoded bitcoin structs to v0.32

### PR DESCRIPTION
**Pull request by tvolk131**
Part of #5200

The primary changes are in `fedimint-core/src/encoding/btc.rs`. The rest of the diffs in this PR are downstream changes.

### CODE REVIEW

**Review by joschisan — DISMISSED** [review]

File: fedimint-core/src/bitcoin_migration.rs [hunk 1]
```diff
@@ -111,6 +119,192 @@ pub fn bitcoin30_to_bitcoin32_address(address: &bitcoin30::Address) -> bitcoin::
...
+    )
+    .expect("Failed to convert bitcoin30 witness to bitcoin32 witness")
+}
+
+fn bitcoin32_to_bitcoin30_witness(witness: &bitcoin::Witness) -> bitcoin30::Witness {
+    bincode::deserialize(
+        &bincode::serialize(witness).expect("Failed to serialize bitcoin32 witness"),
+    )
+    .expect("Failed to convert bitcoin32 witness to bitcoin30 witness")
+}
```
[the remark this answers is not in the document]
Comment (elsirion) [reply fedimint-core/src/bitcoin_migration.rs:184]: Consensus encoding might not be available for both unless we duplicate the impls. But there typically is a from/to basic types impl in rust-bitcoin, e.g. here:

Screenshot 2024-10-30 at 06-11-52 Witness in bitcoin - Rust

[the remark this answers is not in the document]
Comment (joschisan) [reply fedimint-core/src/bitcoin_migration.rs:184]: I meant bitcoin::consensus::encode::Decodable and bitcoin::consensus::encode::Encodable, not our encoding.

[the remark this answers is not in the document]
Comment (elsirion) [reply fedimint-core/src/bitcoin_migration.rs:184]: Oh, right :laughing:

[the remark this answers is not in the document]
Comment (tvolk131) [reply fedimint-core/src/bitcoin_migration.rs]: Fixed for `Witness` and for `PartialMerkleTree`.

**Review by joschisan — APPROVED** [review]

**Pull request merged**